### PR TITLE
Master teardown

### PR DESCRIPTION
**Pull request by mulkieran**

### CODE REVIEW

**Comment by mulkieran** [comment]
@agrover @trgill review? This is necessarily incomplete. But it will have to be implemented completely, and it will be useful for testing reconstruction from metadata.

File: src/engine/structures.rs [hunk 1]
```diff
@@ -33,6 +33,11 @@ impl<T: HasName + HasUuid> Table<T> {
 
     }
 
+    /// Empty this table of all its items, returning them in a vector.
+    pub fn empty(mut self) -> Vec<T> {
+        self.items.drain(..).collect()
```
Comment (agrover) [inline src/engine/structures.rs]: Does it let you skip the drain/collect and just do `self.items` here?

File: src/engine/strat_engine/pool.rs [hunk 2]
```diff
@@ -199,6 +199,21 @@ impl StratPool {
             }
         }
     }
+
+    /// Teardown a pool.
+    /// Take down the device mapper devices belonging to the pool.
+    /// This method and destroy() must keep their DM teardown operations
+    /// in sync.
+    pub fn teardown(self) -> EngineResult<()> {
+        // TODO: sync files
```
Comment (agrover) [inline src/engine/strat_engine/pool.rs]: Not sure -- we may find all filesystems need to be unmounted to teardown, not just synced.

File: src/engine/structures.rs (hunk 1 above)
Comment (mulkieran) [inline src/engine/structures.rs]: yes, it does. all that cleverness for nothing.

**Comment by mulkieran** [comment]
Made changes based on comments. @trgill review?

**Comment by trgill** [comment]
I think you need to run rustfmt on simple_tests.rs before merge.

**Pull request merged**